### PR TITLE
Support for Xcode 11 attributes and objects.

### PR DESCRIPTION
**Pull request by dnkoutso**
Closes https://github.com/CocoaPods/CocoaPods/issues/8871
Closes https://github.com/CocoaPods/Xcodeproj/issues/688

### CODE REVIEW

File: lib/xcodeproj/constants.rb [hunk 1]
```diff
@@ -126,6 +126,7 @@ module Constants
     # @return [Hash] The compatibility version string for different object versions.
     #
     COMPATIBILITY_VERSION_BY_OBJECT_VERSION = {
+      52 => 'Xcode 11.0',
```
Comment (segiddins) [inline lib/xcodeproj/constants.rb]: can we actually load an object version 52 project?

Comment (dnkoutso) [reply lib/xcodeproj/constants.rb]: yes we can?

Comment (dnkoutso) [reply lib/xcodeproj/constants.rb]: I mean I successfully integrated and built an Xcode 11 project.

Comment (segiddins) [reply lib/xcodeproj/constants.rb]: Did you verify it had object version 52? Something must've changed in the project format, so there's got to be something else we need to update

**Comment by jefhai** [comment]
@dnkoutso is this related https://github.com/CocoaPods/Xcodeproj/issues/688 ?

File: lib/xcodeproj/constants.rb (hunk 1 above)
Comment (dnkoutso) [reply lib/xcodeproj/constants.rb]: It is 52 and you are right https://github.com/CocoaPods/Xcodeproj/issues/688

**Comment by dnkoutso** [comment]
from SPM
```
[!] [!] Xcodeproj doesn't know about the following attributes {"productRef"=>"B789131C22A6CEF80033A94E"} for the 'PBXBuildFile' isa.
If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new

[!] [!] Xcodeproj doesn't know about the following attributes {"productRef"=>"B789131C22A6CEF80033A94E"} for the 'PBXBuildFile' isa.
If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new

[!] [!] Xcodeproj doesn't know about the following attributes {"packageProductDependencies"=>["B789131C22A6CEF80033A94E"]} for the 'PBXNativeTarget' isa.
If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new

[!] [!] Xcodeproj doesn't know about the following attributes {"packageReferences"=>["B789131B22A6CEF80033A94E"]} for the 'PBXProject' isa.
If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new
```

**Comment by jefhai** [comment]
@dnkoutso is SPM a tool that shows that information? Can you link me to it?

**Comment by dnkoutso** [comment]
SPM = Swift Package Manager.

**Comment by dnkoutso** [comment]
PR still WIP but added all SPM references.

File: lib/xcodeproj/constants.rb (hunk 1 above)
Comment (dnkoutso) [reply lib/xcodeproj/constants.rb]: @segiddins done.

**Comment by Coeur** [comment]
Don't know if it's related, but some users are reporting issues to find a simulator with `pod lib lint`. See https://github.com/CocoaPods/CocoaPods/issues/8163#issuecomment-498944110

**Comment by dnkoutso** [comment]
@segiddins ready for review. Uncertain on how to update test fixtures.

File: lib/xcodeproj/project/object.rb [hunk 2]
```diff
@@ -518,6 +518,8 @@ def inspect
 end
 
 # Now load the concrete subclasses.
+require 'xcodeproj/project/object/swift_package_remote_reference'
```
Comment (segiddins) [inline lib/xcodeproj/project/object.rb]: nit: sort

File: lib/xcodeproj/project/object/root_object.rb [hunk 3]
```diff
@@ -76,9 +80,19 @@ def ascii_plist_annotation
           ' Project object '
         end
 
+        def to_hash_as(method = :to_hash)
+          hash_as = super
+          if !hash_as['packageReferences'].nil? && hash_as['packageReferences'].empty?
```
Comment (dnkoutso) [inline lib/xcodeproj/project/object/root_object.rb]: @segiddins these values are "optional" as in a default Xcode 11 project will *not* include them in its plist. Is there a better way in Xcodeproj to handle this? 

The way I have it here is passing all tests but felt its a bit ugly. Alternatively, we can update all tests to ensure `packageReferences => []` is taken into consideration even though they are none in the actual `.pbxproj.`

**Comment by benasher44** [comment]
There's also a new one called `inputFiles`. Just saw it crop up this morning in our codebase. It's part of `PXBuildRule`

File: lib/xcodeproj/project/object/native_target.rb [hunk 4]
```diff
@@ -622,6 +627,22 @@ def sort(_options = nil)
...
+          end
+          hash_as
+        end
+
+        def to_ascii_plist
+          plist = super
+          if !plist.value['packageProductDependencies'].nil? && plist.value['packageProductDependencies'].empty?
+            plist.value.delete('packageProductDependencies')
+          end
+          plist
```
Comment (amorde) [inline lib/xcodeproj/project/object/native_target.rb]: is this covered by any tests?

**Comment by dnkoutso** [comment]
@benasher44 will update.

File: lib/xcodeproj/project/object/native_target.rb (hunk 4 above)
Comment (dnkoutso) [reply lib/xcodeproj/project/object/native_target.rb]: it is already covered by existing tests but I might add an explicit one.

**Comment by dnkoutso** [comment]
Added missing `input_files` in `PBXBuildRule`

**Comment by benasher44** [comment]
input_files lg2m

File: lib/xcodeproj/project/object/native_target.rb (hunk 4 above)
Comment (dnkoutso) [reply lib/xcodeproj/project/object/native_target.rb]: added explicit tests.

**Comment by dnkoutso** [comment]
Will release 1.10.0

**Comment by dnkoutso** [comment]
Shipped. `gem update xcodeproj` to get latest version for initial (and maybe final) support for Xcode 11.

**Pull request merged**